### ISSUE TITLE
Take in Secrets Manager Secret ARN for wr.db.get_engine parameters

### ISSUE THREAD

**Issue by kinghuang** [awslabs/aws-data-wrangler#402]
**Is your idea related to a problem? Please describe.**

The [awswrangler.db.get_engine](https://aws-data-wrangler.readthedocs.io/en/latest/stubs/awswrangler.db.get_engine.html#awswrangler.db.get_engine) function creates a SQLAlchemy Engine for the given host, port, user, password, etc. My code gets these parameters by reading a secret (specified by ARN). It would be nice if I could just pass a secret ARN to awswrangler, instead of having to get it myself and unpack all the values for the `get_engine` function.

**Describe the solution you'd like**

Be able to pass a secret ARN to the `get_engine` function. It would extract default values for the `db_type`, `host`, `port`, `database`, `user`, and `password` arguments from it. If any of the arguments are specified, they would override the values from the secret.

**Comment by igorborgest** [comment]
Hey @kinghuang, thanks for reaching out!

Well, what about store your connections attributes into the Glue Catalog Connections? Then you can use [wr.catalog.get_engine()](https://aws-data-wrangler.readthedocs.io/en/latest/stubs/awswrangler.catalog.get_engine.html) like in [this tutorial](https://github.com/awslabs/aws-data-wrangler/blob/master/tutorials/007%20-%20Redshift%2C%20MySQL%2C%20PostgreSQL.ipynb).

You can keep your connection stored there even when it will be used outside glue jobs. We prefer the glue catalog than secret manager in this case because we don't need to rely in an arbitrary attribute structure like an json stringfied. In addition to that the Glue Catalog will also give you the option to test tour connections directly from there, it is quite useful.

**Comment by fitzoh** [comment]
>Well, what about store your connections attributes into the Glue Catalog Connections? Then you can use wr.catalog.get_engine() like in this tutorial.

Just to clarify, should the connection information be stored as JDBC credentials?

I'm pretty sure that's the case, but I didn't find that explicitly stated in the docs, and it's a little counterintuitive (JDBC != python).

On a related note, would the glue catalog approach work for RDS proxy connections?

**Comment by kinghuang** [comment]
We have some connection info in the Glue catalog, too. However, RDS, Redshift, and DocumentDB credentials stored in AWS Secrets Manager get automatic [password rotation](https://docs.aws.amazon.com/secretsmanager/latest/userguide/rotating-secrets.html). There's a defined format for the data that includes details like the engine, host, port, username, password, etc.

I tend to have all the master credentials stored in the AWS Secrets Manager. It wouldn't make sense to place them in Glue.

**Comment by igorborgest** [comment]
Quick note about our fresh new version [2.0.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/2.0.0):

### Breaking changes

* `sqlalchemy` and `psycopg2` dependencies replaced by [`redshift_connector`](https://github.com/aws/amazon-redshift-python-driver) and `pg8000`
* All `wr.db.*` functions was distributed into [`wr.redshift.*`](https://aws-data-wrangler.readthedocs.io/en/latest/api.html#amazon-redshift), [`wr.postgresql.*`](https://aws-data-wrangler.readthedocs.io/en/latest/api.html#postgresql) and [`wr.mysql.*`](https://aws-data-wrangler.readthedocs.io/en/latest/api.html#mysql) ([Tutorial](https://github.com/awslabs/aws-data-wrangler/blob/master/tutorials/007%20-%20Redshift%2C%20MySQL%2C%20PostgreSQL.ipynb))
* Redshift COPY and UNLOAD function was refactored into [`wr.redshift.*`](https://aws-data-wrangler.readthedocs.io/en/latest/api.html#amazon-redshift) ([Tutorial](https://github.com/awslabs/aws-data-wrangler/blob/master/tutorials/008%20-%20Redshift%20-%20Copy%20%26%20Unload.ipynb))
* `wr.catalog.get_engine()` was replaced by [`wr.redshift.connect()`](https://aws-data-wrangler.readthedocs.io/en/latest/stubs/awswrangler.redshift.connect.html), [`wr.postgresql.connect()`](https://aws-data-wrangler.readthedocs.io/en/latest/stubs/awswrangler.postgresql.connect.html), [`wr.mysql.connect()`](https://aws-data-wrangler.readthedocs.io/en/latest/stubs/awswrangler.mysql.connect.html) ([Tutorial](https://github.com/awslabs/aws-data-wrangler/blob/master/tutorials/007%20-%20Redshift%2C%20MySQL%2C%20PostgreSQL.ipynb))

**Comment by igorborgest** [comment]
@kinghuang let's address that in the version `2.1.0`.

What format do you prefer store the connection properties inside secret manager? Regular JSON?

**Comment by fitzoh** [comment]
The secrets manager has a setup wizards for RDS instances and arbitrary non-RDS database instances 
![Screen Shot 2020-12-07 at 1 57 45 PM](https://user-images.githubusercontent.com/131809/101393125-93262d00-3894-11eb-96d1-975840b7fec1.png)



This is the format that is generated if you store a password for an existing RDS instance via the secrets manager GUI
![Screen Shot 2020-12-07 at 1 53 57 PM](https://user-images.githubusercontent.com/131809/101392359-ad134000-3893-11eb-8587-aceb590f39dd.png)

This is the format if you store credentials for a non-rds database:
![Screen Shot 2020-12-07 at 1 56 37 PM](https://user-images.githubusercontent.com/131809/101392765-214de380-3894-11eb-8ab4-c7043d44c923.png)

**Comment by kinghuang** [comment]
I'm having a little trouble finding the docs page for this. But, @fitzoh shows it in the screenshots!

**Comment by igorborgest** [comment]
It is already in a development branch. Could you guys give it a try?

```sh
pip install git+https://github.com/awslabs/aws-data-wrangler.git@feature/secrets-manager
```

e.g.

```py3
 con = wr.redshift.connect(secret_id="aws-data-wrangler/redshift", dbname=dbname)
 df = wr.redshift.read_sql_query("SELECT 1", con=con)
 con.close()
```

> p.s. Also has similar support for `wr.postgresql.connect()` and `wr.mysql.connect()`

> p.p.s Also added support for the `dbname` argument to overwrite the stored one.

**Comment by igorborgest** [comment]
Released on version [2.1.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/2.1.0)
